src/librustc_mir/interpret/traits.rs

```diff
@@ -113,3 +100,41 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         Ok((Size::from_bytes(size), Align::from_bytes(align).unwrap()))
     }
 }
+
+
+#[derive(Clone)]
+pub enum VtableComponent<'tcx> {
+    Usize(u64),
+    Nullptr,
+    FnPtr(Instance<'tcx>),
+}
```

Comment:
Note, in case @oli-obk sees this until tomorrow: I've already told @bjorn3 they can instead use miri to create *the whole* vtable allocation and just convert that in the backend uniformly.

Reply:
Note that this should be done without an `EvalContext`. With my recent changes to `Allocation`s this should be mostly possible. I was hoping on getting https://github.com/rust-lang/rust/pull/55392/files#diff-2eb1005fe45cbe73c0ddfcde017b5f73R83 into master before having to address vtable generation, because that function would make such an operation trivial. You can still look at that function as a reference for creating the function that generates a vtable `Allocation`

Reply:
The code should probably move to `rustc::mir::interpret::traits`, too

Reply:
I am backing out the vtable generation changes for now, because they are a bit too much work for me right now.